tests/Engine/AlgoliaSyncEngine.php

```diff
@@ -15,7 +15,7 @@ public function update($searchableEntities)
         $batch = $this->doUpdate($searchableEntities);
 
         foreach ($batch as $indexName => $response) {
-            $this->algolia->initIndex($indexName)->waitTask($response['taskID']);
+            $this->algolia->waitTask($indexName, $response['taskID']);
```

Comment:
What's the rationale of not using `$this->algolia->index($indexName)->waitTask($response['taskID'])`. It feels weird to have this `waitTask` method taking the index name as 1st parameter.

Reply:
We recently exposed waitTask in the client for 2 main reasons:

1. To make it easier to work with [multiple index endpoint](https://www.algolia.com/doc/rest-api/search/#batch-write-operations-multiple-indices) (which are on the client)
2. To make it easier to work with other API, like analytics that returns a index name and a task ID.

More details: https://github.com/algolia/algoliasearch-client-php/pull/410

I thought it would best to avoid instanciating an index "just" for a waitTask but I didn't benchmark anything, I guess the performance gain would be negligable. It just felt easier.

Note that internally, `Client::waitTask` will instanciate an `Index` until we find a better solution to not duplicate the wait logic, which I don't think is possible with PHP 5.3 because:
- Traits are not available (introduce in 5.4)
- `this` is not available in closures (introduce in 5.4)

Reply:
And thanks for taking the time to review 🙏 Feedback is greatly appreciated!